pytorch_lightning/trainer/trainer.py

```diff
@@ -1002,12 +1002,12 @@ def run_evaluation(self, on_epoch: bool = False) -> _EVALUATE_OUTPUT:
                 ],
             )
 
-        # hook
-        self.evaluation_loop.on_evaluation_end()
-
         # log epoch metrics
         eval_loop_results = self.logger_connector.get_evaluate_epoch_results()
 
+        # hook
+        self.evaluation_loop.on_evaluation_end()
```

Comment:
Some users might be logging in on_evaluation_end and this a BC. Also did you disable the logging in `on_evaluation_end` hook ?

Reply:
`on_evaluation_end` calls either `on_test_end` or `on_validation_end` and they do not support logging (in current master)

See: https://github.com/PyTorchLightning/pytorch-lightning/issues/7166#issuecomment-826808773